libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -383,7 +384,21 @@ export class PackageExtractor {
     }
 
     await this._performExtractionAsync(options, state);
-    await state.assetHandler.finalizeAsync();
+    await state.assetHandler.finalizeAsync({
+      onAfterExtractSymlinksAsync: async () => {
+        // We need the symlinks to be created before attempting to create the bin links, since it requires
+        // the node_modules folder to be realized. While we're here, we may as well perform some specific
+        // link creation tasks and write the extractor-metadata.json file before the asset handler finalizes.
+        if (linkCreation === 'default') {
+          await this._makeBinLinksAsync(options, state);
+        } else if (linkCreation === 'script') {
+          await this._writeCreateLinksScriptAsync(options, state);
```

Comment:
This scenario handles the `.bin` folders correctly?

Reply:
We don't create bin folders for non-default modes, no. This has to do with having the node_modules folder being required in the deployment output folder, which the other mode (script mode) does not create.